walk.go

```diff
@@ -105,8 +108,15 @@ func walkFull(o interface{}, root map[string]interface{}, path llpath.Path, expa
 }
 
 // walkFull walks the given map[string]interface{} tree.
-func walkFullMap(m map[string]interface{}, root map[string]interface{}, p llpath.Path, expandPaths bool, wo walkObserver) (err error) {
-	for k, v := range m {
+func walkFullMap(mVal, rootVal reflect.Value, p llpath.Path, expandPaths bool, wo walkObserver) (err error) {
+	if mVal.Kind() != reflect.Map {
+		return fmt.Errorf("could not walk not map type for %s", mVal)
+	}
+
+	for _, kVal := range mVal.MapKeys() {
+		vVal := reflect.ValueOf(mVal.MapIndex(kVal).Interface())
+		k := kVal.String()
```

Comment:
what if `kVal.Kind() != reflect.String`?

Reply:
Good point. I think that this is where things get weird.

I would like the library to handle arbitrary map types in the future, including non-string keys. My thinking is that this gets us partly there. 

In a future version we could internally represent keys as `fmt.Sprintf("%T:%s", kVal.Interface(), value)`.

Reply:
But to answer your question, the answer for today is that we don't handle that well, but we'd probably do what people expect in most cases.

Reply:
OK, having messed around with this a bit I think we need a runtime check to validate maps that are not string keyed as scalar values. So, you can still validate them, but you need to do it in one isdef, essentially treating it as a scalar value.